File: algorithms/medium/minimum_rounds_to_complete_all_tasks.cpp
class Solution {
public:
  int minimumRounds(vector<int> &tasks) {
    unordered_map<int, int> mp;
    for (auto t : tasks) {
      mp[t]++;
    }

    int ans{0};
    for (auto &[_k, v] : mp) {
      if (v == 1) {
        return -1;
      }
      ans += int(ceil(v / 3.0));
    }
    return ans;
  }
};

File: algorithms/medium/minimum_rounds_to_complete_all_tasks.rs
use std::collections::HashMap;

impl Solution {
    pub fn minimum_rounds(tasks: Vec<i32>) -> i32 {
        let mut mp = HashMap::new();
        for t in tasks {
            *mp.entry(t).or_insert(0) += 1;
        }
        let mut ans = 0;
        for (_, v) in mp {
            if v == 1 {
                return -1;
            } else {
                ans += (v as f64 / 3.0).ceil() as i32;
            }
        }
        ans
    }
}
